doc/developer/design/20230210_storage_persist_sink_redux.md

```diff
@@ -0,0 +1,161 @@
+appended to a shard. We only emit finalized batches to the `append_batches` operator when that batch is contained within
+a batch descriptions that we learned about from `mint_batch_descriptions`.
+3. `append_batches` takes as input the minted batch descriptions and written batches. Whenever the frontiers sufficiently advance,
+we take a batch description and all the batches that belong to it and append it to the persist shard.
+This also updates the shared frontier information in `storage_state` to match the persist frontier.
+This way, the controller eventually learns of the upper frontier of data written to the persist shard.
+Note that failing to `compare_and_append` must be retried, as we current expect only 1 replica to be
+running this sink operation for each source `Collection`.
+
+### Notes on the implementation
+
+- It is needlessly expensive to create a `BatchBuilder` for each timestamp.
+  - A change to this implementation is to avoid spilling data to persist for timestamps
+  whose values use a small amount of memory. These can be consolidated into large batches in `append_batches`,
+  similar to how <https://github.com/MaterializeInc/materialize/pull/17526> works.
+  - We do this in the current implementation, so doing it from many workers, instead of 1,
+  is strictly better.
+  - Its undecided whether or not this should be required for the reference implementation.
+- `append_batches` and `write_batches` will be required to track statistics to expose to users.
+  - See <https://www.notion.so/materialize/User-facing-metrics-for-Sources-9603b61224a04a749969702e04662ed3> for more details
+
+## Alternatives
+
+*List of alternative approaches considered, including pros/cons or why they were rejected.*
+
+- Eliminate the storage `persist_sink` and replace with it with the compute one. Rejected for the following reasons:
+  - Requires skipping self-correction on startup, which is deemed too complex to maintain.
+  - No clear way to keep the self-correction mechanism while also using bounded amounts of memory.
+    - A core design principle of storage is that the ingested collections ("sources") can be
+    arbitrarily sized, and its memory usage should only scale with the size of the largest
+    transaction. The `perist_sink` implementation must conform to this requirement.
+    - Its possible we re-evaluate if the compute sink can use bounded memory in the future,
+    particularly when we have out-of-band consolidation of batches.
+  - Self-correction does additional work that uses compute resources, whereas storage wants to optimize for the
+  raw performance of storage ingestion.
+- Share more of the implementation with the compute `persist_sink`, in particular the `append_batches` operator. Its also
+  possible that `mint_batch_descriptions` could be shared, but the `persist_collection` input would need to be fabricated.
+  Rejected for the following reasons:
```

Comment:
let's be extra clear there that it's only out of scope for the initial impl. I'd imagine that we do want to do this at some point